### ISSUE TITLE
After the recent upgrade to v4.2.0 events are not rendered anymore.

### ISSUE THREAD

**Issue by techouse** [fullcalendar/fullcalendar-vue#32]
Hey!

Great Vue component, however after I upgraded from `v4.1.1` to `v4.2.0` my events are not rendered anymore. What's more I also got some deprecation warnings

```
Use of datesRender as an event is deprecated. Please convert to a prop.
Use of eventRender as an event is deprecated. Please convert to a prop.
Use of viewSkeletonRender as an event is deprecated. Please convert to a prop.
```

Could you please make a working example with `4.2.0` or at least explain what needs to be modified in order for the events to render?

Could it be related to the fact that I initialise `FullCalendar` with an empty `events` array and am then loading my events in Vue's `mounted()` lifecycle hook?

Thanx!

**Comment by doncatnip** [comment]
4.2 seems broken. Resources are not shown anymore, even if I use props instead of an event. It's also undocumented. I'd guess we have to wait a little bit longer :)

**Comment by jameshood118** [comment]
Can confirm, but its not just an upgrade issue, my team switched over to this from a different component, which was a wrapper for fullcalendar). I can get events to render when i manually set an array of objects on vue data, however, our data comes from a graphql query, data is being passed into the component, but is not displaying the events

**Comment by arshaw** [comment]
I will fix this ASAP, but I need a [reduced test case](https://fullcalendar.io/reduced-test-cases). 

Here are some 4.2.0 debug templates I made
- month view: https://codesandbox.io/s/8xyz32l0r8
- timeline view with resources: https://codesandbox.io/s/fullcalendar-vue-w-resources-oqzcg

Both of these examples seem to work with 4.2.0

I need more info from you before i can fix

**Comment by arshaw** [comment]
@techouse whatever you had previous in regards to datesRender/eventRender/viewSkeletonRender should continue to work fine. eventually, I'll want you to write:

```
<FullCalendar :datesRender="myDatesRenderFunc" />
```

@doncatnip, what's undocumented?

**Comment by jameshood118** [comment]
When i take my returned data set, from my graphql query, and set it on data, it renders, but using vue-apollo to hit my graphql endpoint to return data doesnt seem to render. Maybe trying consuming a rest end point?

https://jsonplaceholder.typicode.com/posts

Its certainly not exactly the same things fullcalendar expects to see, but something should display

**Comment by arshaw** [comment]
@jameshood118 , the problem is probably not dependent on graphql. Its seems like a bug related to asynchronously setting the events after initialization. Any way you can simulate that in a recreation? Just set the events to a static array?

**Comment by jameshood118** [comment]
> @jameshood118 , the problem is probably not dependent on graphql. Its seems like a bug related to asynchronously setting the events after initialization. Any way you can simulate that in a recreation? Just set the events to a static array?

See above, i hit post too soon.

**Comment by jameshood118** [comment]
Vue Dev tools, showing the event is being passed into the component, past that, i've no clue.

![Screen Shot 2019-06-03 at 3 15 37 PM](https://user-images.githubusercontent.com/18595004/58828054-862adc80-8612-11e9-99a2-094241edea62.png)

**Comment by arshaw** [comment]
@jameshood118 that helps a little, but I really need a reduced test case to proceed.

**Comment by ColdIce1** [comment]
Here is a test case: https://codesandbox.io/s/fullcalendar-vue-w-resources-tv064

**Comment by jameshood118** [comment]
Maybe not the best example, but:

https://codesandbox.io/s/fullcalendar-vue-e02ox

using the jsonplaceholder i previously mentioned, this console logs the return, but its also using it in the calendar (which doesnt render)

**Comment by arshaw** [comment]
@ColdIce1, thanks for the recreation. I now see the problem. It's a bug with accepting functions for events/resources. The new algorithm relies on recursively copying some of the props, and that doesn't work for functions. As soon as I'm confident there's no other causes for this bug and related bugs, I'll get to work on a fix.

@jameshood118, it looks like you're hoping the JSON at https://jsonplaceholder.typicode.com/todos/1 would render events, but it's not a valid array of valid event objects, so it won't render anyway.

**Comment by jameshood118** [comment]
You are right, I missed that part, my apologies

**Comment by chrisribal** [comment]
I can also confirm that this bug seems to also affect the resources. In my application i'm loading resources and events via AJAX, but no event nor resource get's rendered in 4.2.0.

**Comment by arshaw** [comment]
@chrisribal can you post a recreation based on this?
https://codesandbox.io/s/fullcalendar-vue-w-resources-oqzcg

i need to see if you're using a function as your `resources` setting or not. or maybe something else is going on

**Comment by doncatnip** [comment]
> @doncatnip, what's undocumented?

Ho to use that props exactly. I guess it is on the fullcalendar docs itself, but not here in the vue wrapper nor the vue wrapper example. The tag message doesn't say anything about an api change either.

**Comment by doncatnip** [comment]
The problem seems to be with computed resources: if first an empty Array is returned, then any subsequent arrays filled with actual resources will not be shown:
https://codesandbox.io/s/fullcalendar-vue-w-resources-gn84y

**Comment by chrisribal** [comment]
@doncatnip @arshaw That's exactly the problem. For testing I passed an empty array to the resources prop and called a window.setTimeout to fill that array with resources. But the changes do not take effect and the resources never appear.

**Comment by arshaw** [comment]
I just released [4.2.1](https://github.com/fullcalendar/fullcalendar-vue/releases/tag/v4.2.1). I believe I fixed everything. Thanks for the bug reports.

Could you all PLEASE CONFIRM this release fixes your issues?

**Comment by jameshood118** [comment]
Pulling and testing now

**Comment by jameshood118** [comment]
YAY i have events on my calendar, you're the man @arshaw .. Well you and your team, obviously

**Comment by zintj** [comment]
@arshaw Seemed to fix all the issues that popped up for me in 4.2.0

**Comment by arshaw** [comment]
I'm glad everything is seems to be fixed.

I needed to publish another patch release 4.2.2 because I noticed another bug related to event objects with Date objects. **Please upgrade to 4.2.2**.

Closing this issue. Please open a new ticket if you experience continued problems.

**Comment by techouse** [comment]
All fixed for me as well. Thanx ❤️

**Comment by jaythegeek** [comment]
I seem to get this issue when I load from a json URL, I have tried various different things to fix this. I have run it as an identical copy of the example. I start with `calendarEvents: []` and then populate that from an axios get request where I quite simply do `this.calendarEvents = data` when I use dev tools I can see the Array of objects with my events properly filled out, but they do not render on the calendar, I have tried using the feed in eventResources too but to no avail, the call is made successfully but the events are not populated. I will be back in abit and will post this as a reduced-test-case but wondered if anyone had any ideas in the meantime as this is impacting a fairly significant deploy.
Thanks in advance
